src/catchup/ApplyBucketsWork.cpp

```diff
@@ -77,6 +77,14 @@ ApplyBucketsWork::onReset()
 
     if (!isAborting())
     {
+        // clear ledgerTxn state of all ledger entries in preparation of bucket
+        // apply
+        if (!mApp.getConfig().MODE_USES_IN_MEMORY_LEDGER)
+        {
+            auto& lsRoot = mApp.getLedgerTxnRoot();
+            lsRoot.deleteObjectsModifiedOnOrAfterLedger(0);
```

Comment:
I don't think we want to do this if the work fails?

Reply:
yes we want to always reset on failure: otherwise we leave the ledger in a partially applied state (which is non obvious when debugging, when this could happen)